crates/noirc_frontend/src/hir/def_collector/errors.rs

```diff
@@ -29,6 +30,19 @@ pub enum DefCollectorErrorKind {
     NonStructTypeInImpl { span: Span },
     #[error("Cannot `impl` a type defined outside the current crate")]
     ForeignImpl { span: Span, type_name: String },
+    #[error("Mismatch signature of trait")]
+    MismatchTraitSignature { primary_message: String, secondary_message: String, span: Span },
```

Comment:
This error is too general, lets split it up into more specific variants like the errors below are.

Reply:
I agree. Removing this error